provider/src/main/kotlin/org/gradle/kotlin/dsl/DependencyHandlerScope.kt

```diff
@@ -39,9 +39,76 @@ class DependencyHandlerScope(val dependencies: DependencyHandler) : DependencyHa
     operator fun String.invoke(dependencyNotation: Any): Dependency =
         dependencies.add(this, dependencyNotation)
 
+    /**
+     * Adds a dependency to the given configuration.
+     *
+     * @param dependencyNotation notation for the dependency to be added.
+     * @param dependencyConfiguration expression to use to configure the dependency.
+     * @return The dependency.
+     * @see DependencyHandler.add
+     */
+    inline
     operator fun String.invoke(dependencyNotation: String, dependencyConfiguration: ExternalModuleDependency.() -> Unit): ExternalModuleDependency =
         dependencies.add(this, dependencyNotation, dependencyConfiguration)
 
+    /**
+     * Adds a dependency to the given configuration.
+     *
+     * @param group the group of the module to be added as a dependency.
+     * @param name the name of the module to be added as a dependency.
+     * @param version the optional version of the module to be added as a dependency.
+     * @param configuration the optional configuration of the module to be added as a dependency.
+     * @param classifier the optional classifier of the module artifact to be added as a dependency.
+     * @param ext the optional extension of the module artifact to be added as a dependency.
+     * @return The dependency.
+     *
+     * @see DependencyHandler.add
+     */
+    operator fun String.invoke(
+        group: String,
+        name: String,
+        version: String? = null,
+        configuration: String? = null,
+        classifier: String? = null,
+        ext: String? = null): ExternalModuleDependency =
+        dependencies.create(group, name, version, configuration, classifier, ext).apply { add(this@invoke, this) }
+
+    /**
+     * Adds a dependency to the given configuration.
+     *
+     * @param group the group of the module to be added as a dependency.
+     * @param name the name of the module to be added as a dependency.
+     * @param version the optional version of the module to be added as a dependency.
+     * @param configuration the optional configuration of the module to be added as a dependency.
+     * @param classifier the optional classifier of the module artifact to be added as a dependency.
+     * @param ext the optional extension of the module artifact to be added as a dependency.
+     * @param dependencyConfiguration expression to use to configure the dependency.
+     * @return The dependency.
+     *
+     * @see DependencyHandler.create
+     * @see DependencyHandler.add
+     */
+    inline
```

Comment:
Why is this method inline where the others are not?

Reply:
This method is not the only one that is not inlined. It's not a higher-order function so it doesn't need to be, see https://kotlinlang.org/docs/reference/inline-functions.html